services/datamanager/builtin/capture/capture.go

```diff
@@ -124,6 +138,75 @@ func (c *Capture) newCollectors(collectorConfigsByResource CollectorConfigsByRes
 	return newCollectors
 }
 
+func (c *Capture) mongoSetup(ctx context.Context, mc MongoConfig) *mongo.Collection {
+	if err := mc.validate(); err != nil {
+		c.logger.Warn("mongo_config is invalid: %s", err.Error())
+		return nil
+	}
+	// closes collectors and old connection if there is one and nils out mongo connection struct members
+	c.closeNoMongoMutex(ctx)
+	// Use the SetServerAPIOptions() method to set the Stable API version to 1
+	serverAPI := options.ServerAPI(options.ServerAPIVersion1)
```

Comment:
What does this do? I'm not familiar

Reply:
I imagine that it sets the API version to use when the client connects to mongo. 

I got it from this connection guide: https://www.mongodb.com/docs/drivers/go/current/fundamentals/connections/connection-guide/

Do you think it is not necessary?

Reply:
Hm, I've never seen it used before. According to the doc, this is its purpose:

```You can set the Stable API version as an option to avoid breaking changes when you upgrade to a new server version. To learn more about the Stable API feature, see the [Stable API page.](https://www.mongodb.com/docs/drivers/go/current/fundamentals/stable-api/#std-label-golang-stable-api)```

I'm unclear on exactly how it helps you avoid those breaking changes. @benjirewis hope you don't mind me looping you in here, but do you know anything about this feature?

Reply:
I do yeah; it's to specify a version of the API you'd like to use in communicating between the Go driver and an MDB instance. Basically, users were frustrated by there being different semantic versions for drivers (SDKs) and MDB servers and having to keep track of compatibility as a matrix, so we introduced a single "API" version that a driver could "ask" for, and the server, even if it was much more modern than the driver, could contort itself to behave like an older server. It certainly won't hurt to specify it here, but I also don't think it's totally necessary.

Reply:
Thanks! I'm inclined to leave it as it is in the connection guide.